platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/implementing_gradle_plugins_binary.adoc

```diff
@@ -0,0 +1,621 @@
+These are known as *conventions*.
+
+Conventions are properties that are initialized with default values and can be overridden by the user in their build script.
+For example:
+
+====
+include::sample[dir="snippets/customPlugins/customPluginWithConvention/kotlin",files="build.gradle.kts[]"]
+include::sample[dir="snippets/customPlugins/customPluginWithConvention/groovy",files="build.gradle[]"]
+====
+
+----
+$ gradle -q hello
+include::{snippetsPath}/customPlugins/customPluginWithConvention/tests/customPluginWithConvention.out[]
+----
+
+In this example, `GreetingPluginExtension` is a class that represents the convention.
+The message property is the convention property with a default value of 'Hello from GreetingPlugin'.
+
+Users can override this value in their build script:
+
+====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+GreetingPluginExtension {
+    message = "Custom message"
+}
+----
+=====
+[.multi-language-sample]
+=====
+.build.gradle
+[source,groovy]
+----
+GreetingPluginExtension {
+    message = 'Custom message'
+}
+----
+=====
+====
+
+[source,text]
+----
+$ gradle -q hello
+Custom message
+----
+
+[[capabilities-vs-conventions]]
+== Separating capabilities from conventions
+
+Separating capabilities from conventions in plugins allows users to choose which tasks and conventions to apply.
+
+For example, the Java Base plugin provides un-opinionated functionality like `SourceSets`, while the Java plugin adds tasks and conventions familiar to Java developers like `classes`, `jar` or `javadoc`.
+
+When designing your own plugins, consider developing two plugins — one for capabilities and another for conventions — to offer flexibility to users.
+
+In the example below, `MyPlugin` contains conventions, and `MyBasePlugin` defines capabilities.
+Then, `MyPlugin` applies `MyBasePlugin`, this is called _plugin composition_.
+To apply a plugin from another one:
+
+.MyBasePlugin.java
+[source,java]
+----
+include::{snippetsPath}/developingPlugins/capabilitiesVsConventions/groovy/buildSrc/src/main/java/MyBasePlugin.java[]
+----
+
+.MyPlugin.java
+[source,java]
+----
+include::{snippetsPath}/developingPlugins/capabilitiesVsConventions/groovy/buildSrc/src/main/java/MyPlugin.java[]
+----
+
+== Reacting to plugins
+
+A common pattern in Gradle plugin implementations is configuring the runtime behavior of existing plugins and tasks in a build.
+
+For example, a plugin could assume that it is applied to a Java-based project and automatically reconfigure the standard source directory:
+
+[[convention-api-usage-example]]
+.InhouseStrongOpinionConventionJavaPlugin.java
+[source,java]
+----
+include::{snippetsPath}/developingPlugins/reactingToPlugins/groovy/buildSrc/src/main/java/InhouseStrongOpinionConventionJavaPlugin.java[tags=snippet]
+----
+
+The drawback to this approach is that it automatically forces the project to apply the Java plugin, imposing a strong opinion on it.
```

Comment:
❓ What does "imposing a strong opinion on it" mean?  This should be explained further.